legion/engine/core/logging/logging.hpp

```diff
@@ -292,22 +297,21 @@ namespace legion::core::log
     };
 
 
+#define LOG_FILE impl::log_file
+#define logger impl::logger
 
-#if defined(LEGION_KEEP_CONSOLE) || defined(LEGION_DEBUG)
-
-#define LOG_FILE "stdout"
-#define logger impl::console_logger
-
-#else
-
-#define LOG_FILE "logs/args-engine.log"
-#define logger impl::file_logger
-
-#endif
 
     /** @brief sets up logging (do not call, invoked by engine) */
     inline void setup()
     {
+#if defined(LEGION_KEEP_CONSOLE) || defined(LEGION_DEBUG)
+        LOG_FILE = impl::console_file;
+        logger = impl::console_logger;
+#else
```

Comment:
this is very likely to cause problems when events are logged before the engine is started up. I would recommend settting the default logger to be the console logger at first and replacing it with the file logger if necessary